File: backend/src/models/ExplorePlant.model.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IExplorePlant extends Document {
  name: string;
  scientificName: string;
  category: string;
  difficulty: 'Easy' | 'Moderate' | 'Difficult';
  light: string;
  water: string;
  description: string;
  tags: string[];
  icon?: string;
  imageUrl?: string;
  isActive: boolean;
  createdAt: Date;
  updatedAt: Date;
}

const ExplorePlantSchema = new Schema<IExplorePlant>(
  {
    name: {
      type: String,
      required: true,
      trim: true,
      unique: true,
    },
    scientificName: {
      type: String,
      required: true,
      trim: true,
    },
    category: {
      type: String,
      required: true,
      enum: ['Indoor', 'Outdoor', 'Low Maintenance', 'Pet Safe', 'Flowering'],
      trim: true,
    },
    difficulty: {
      type: String,
      required: true,
      enum: ['Easy', 'Moderate', 'Difficult'],
      default: 'Easy',
    },
    light: {
      type: String,
      required: true,
      trim: true,
    },
    water: {
      type: String,
      required: true,
      trim: true,
    },
    description: {
      type: String,
      required: true,
      trim: true,
    },
    tags: {
      type: [String],
      default: [],
    },
    icon: {
      type: String,
      trim: true,
    },
    imageUrl: {
      type: String,
      trim: true,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
  },
  {
    timestamps: true,
  }
);

// Indexes
ExplorePlantSchema.index({ category: 1, isActive: 1 });
ExplorePlantSchema.index({ name: 'text', scientificName: 'text', description: 'text' });

export const ExplorePlant = mongoose.model<IExplorePlant>('ExplorePlant', ExplorePlantSchema);

File: frontend/lib/features/explore/data/models/explore_plant_model.dart
import '../../domain/entities/explore_plant.dart';

class ExplorePlantModel extends ExplorePlant {
  const ExplorePlantModel({
    required super.id,
    required super.name,
    required super.scientificName,
    required super.category,
    required super.difficulty,
    required super.light,
    required super.water,
    required super.description,
    required super.tags,
    super.icon,
    super.imageUrl,
  });

  factory ExplorePlantModel.fromJson(Map<String, dynamic> json) {
    return ExplorePlantModel(
      id: json['id'] as String,
      name: json['name'] as String,
      scientificName: json['scientificName'] as String,
      category: json['category'] as String,
      difficulty: json['difficulty'] as String,
      light: json['light'] as String,
      water: json['water'] as String,
      description: json['description'] as String,
      tags: (json['tags'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
      icon: json['icon'] as String?,
      imageUrl: json['imageUrl'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'scientificName': scientificName,
      'category': category,
      'difficulty': difficulty,
      'light': light,
      'water': water,
      'description': description,
      'tags': tags,
      'icon': icon,
      'imageUrl': imageUrl,
    };
  }
}
